mmdet/models/detectors/semi_base.py

```diff
@@ -76,12 +76,13 @@ def loss(self, multi_batch_inputs: Dict[str, Tensor],
         losses = dict()
         losses.update(**self.loss_by_gt_instances(
             multi_batch_inputs['sup'], multi_batch_data_samples['sup']))
-        origin_pseudo_instances, batch_info = self.get_pseudo_instances(
-            multi_batch_inputs['unsup_teacher'],
-            multi_batch_data_samples['unsup_teacher'])
+        with torch.no_grad():
+            origin_pseudo_data_samples, batch_info = self.get_pseudo_instances(
```

Comment:
The function name seems unreasonable, `get_pseudo_instances` actually returns datasample. The same  API deals with datasample of `project_pseudo_instances` and `loss_by_pseudo_instances`